src/file_name.rs

```diff
@@ -0,0 +1,42 @@
+use serde::{Deserialize, Serialize};
+use std::{
+    ffi::{CString, OsStr},
+    os::unix::ffi::OsStrExt,
+    path::Path,
+};
+use zbus::zvariant::Type;
+
+/// A file name represented as a null-terminated byte array.
+///
+/// Implements `AsRef<Path>`.
```

Comment:
No need for this part I believe. it is already in the docs. 

Also please move this file to the src/ directory as it is common with various "sub-directories" and would also be used by the backend.